src/scry/implementations.cr

```diff
@@ -34,10 +34,34 @@ module Scry
 
     # NOTE: compiler is a bit heavy in some projects.
     def search(filename, position)
+      local_require = get_local_require(filename, position)
+      return local_require if local_require
       scope = get_scope(@workspace.root_uri)
       analyze(filename, position, scope)
     end
 
+    LOCAL_REQUIRE_REGEX = /^require\s+"(\.{1,2}.*?)"\s*$/
+    POSITION_0          = Protocol::Position.new(0, 0)
+    RANGE_0             = Protocol::Range.new(POSITION_0, POSITION_0)
+
+    private def get_local_require(filename, position)
+      line = @text_document.get_line(position.line)
+      if match = line.match(LOCAL_REQUIRE_REGEX)
+        requirement = match[1]
+        start_pos = line.index(requirement)
+        return unless start_pos
+        end_pos = start_pos + requirement.size - 1
+        if position.character > start_pos || position.character < end_pos
+          filepath = "#{File.expand_path(requirement, File.dirname(filename))}.cr"
```

Comment:
Iirc one could write `require "./foo.cr"`, so you should first check the expanded requirement without adding the extension

Reply:
Maybe sth like:
```cr
# a little helper
def return_if_exists?(path)
  File.exists?(path) && path
end

expanded_req = File.expand_path(requirement, File.dirname(filename))
filepath = return_if_exists?(expanded_req) || return_if_exists?("#{expanded_req}.cr")
if filepath
  # ... 
end
```